### PR TITLE
Got working version of calculator, no map

### PR DESCRIPTION
**Pull request by buuck**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,17 +1,286 @@
...
+import duckdb
+import pandas as pd
+import datetime
 import ee
 # import geemap
+import yaml
+
+# Define constants
+MD_SERVICE_TOKEN = 'md_service_token.txt'
+# to-do: set-up with papermill parameters
```
Comment (pgzmnk) [inline app.py:25]: Papermill will no longer be used. It was meant for the scrapped Airflow methodology.

File: app.py [hunk 2]
```diff
@@ -1,17 +1,286 @@
...
+YEAR = 2020
+LOCATION=[-74.653370, 5.845328]
+ROI_RADIUS = 20000
+GEE_SERVICE_ACCOUNT = 'climatebase-july-2023@ee-geospatialml-aquarry.iam.gserviceaccount.com'
+GEE_SERVICE_ACCOUNT_CREDENTIALS_FILE = 'ee_service_account.json'
+INDICES_FILE = 'indices.yaml'
+START_YEAR = 2015
+END_YEAR = 2022
+
+class IndexGenerator:
```
Comment (pgzmnk) [inline app.py:36]: We should place `IndexGenerator` in its own file. e.g. `utils/index_generator.py`.

File: app.py [hunk 3]
```diff
@@ -1,17 +1,286 @@
...
+            map.addLayer(dataset, index_config['viz'], index_config['name'])
+        print(f"Generated index: {index_config['name']}")
+        return dataset
+
+    def zonal_mean_index(self, index_key):
+        index_config = self.indices[index_key]
+        dataset = self.generate_index(index_config)
+        # zm = self._zonal_mean(single, index_config.get('bandname') or 'constant')
+        out = dataset.reduceRegion(**{
+                'reducer': ee.Reducer.mean(),
```
Comment (pgzmnk) [inline app.py:150]: Normalize each index to a value between 0 and 100 based on min/max for that layer. `ee.Reducer.minMax()` can help here.

File: app.py [hunk 4]
```diff
@@ -1,17 +1,286 @@
...
+                'geometry': self.roi,
+                'scale': 200 # map scale
+            }).getInfo()
+        if index_config.get('bandname'):
+            return out[index_config.get('bandname')]
+        return out
+
+    def generate_composite_index_df(self, indices=[]):
+        data={
```
Comment (pgzmnk) [inline app.py:159]: To-do:
1. Introduce `coefficient` field to this json that turns into a dataframe.
2. Introduce in `indices.yaml` coefficients for each component layer of the index.
3. Introduce coefficients in score calculation them . e.g. `Index = C_1 * Layer_1 + C_2 * Layer_2 ...`.

**Comment by pgzmnk** [comment]
Merging in order to advance development.

Left comments. They should be addressed in follow-up PR.

**Comment by pgzmnk** [comment]
Also, huggingface plays better with pip than conda. I'll make the change.

**Comment by buuck** [comment]
> Also, huggingface plays better with pip than conda. I'll make the change.

That's fine; I just like to have a conda environment available so that I can use it as a Jupyter kernel. But that doesn't necessarily have to be in the repo.

**Pull request merged**